raiden/tests/integration/long_running/test_integration_events.py

```diff
@@ -501,7 +512,7 @@ def run_test_secret_revealed_on_chain(
     # intercepting it and app2 has not received the updated balance proof
 
     # Close the channel. This must register the secret on chain
-    channel_close_event = ContractSendChannelClose(
+    channel_close_event = ContractSendChannelClose(  # type: ignore
```

Comment:
Is this `type: ignore`d because `on_raiden_event` accepts `Event` only? Another alternative is to do

```python
channel_close_event: Event = ...
```

Reply:
The problem is
```
Argument "balance_proof" to "ContractSendChannelClose" has incompatible type "Union[BalanceProofSignedState, BalanceProofUnsignedState, None]"; expected "Optional[BalanceProofSignedState]"
```
 
I didn't want to pull that out and do asserts on it, since we probably want to get rid of the `Union` and always use signed BPs, anyway. However, doing that involves more changes than just the type declaration in `NettingChannelEndState`, so I've ignore that for this PR.